lib/Backend/BackwardPass.cpp

```diff
@@ -1608,6 +1615,84 @@ BackwardPass::OptBlock(BasicBlock * block)
             intOverflowDoesNotMatterInRangeBySymId->ClearAll();
         }
     }
+
+#if DBG
+    if (this->DoByteCodeUpwardExposedUsed())
+    {
+        if (this->tag == Js::BackwardPhase)
+        {
+            // Keep track of all the bytecode upward exposed after Backward's pass
+            BVSparse<JitArenaAllocator>* byteCodeUpwardExposedUsed = nullptr;
+            const auto EnsureBV = [&] { if (!byteCodeUpwardExposedUsed) byteCodeUpwardExposedUsed = JitAnew(this->globOpt->alloc, BVSparse<JitArenaAllocator>, this->globOpt->alloc); };
+            Assert(block->byteCodeUpwardExposedUsed != nullptr);
+            FOREACH_BITSET_IN_SPARSEBV(symID, block->byteCodeUpwardExposedUsed)
+            {
+                Sym* sym = this->func->m_symTable->Find(symID);
+                if (sym != nullptr && sym->IsStackSym())
+                {
+                    StackSym* stackSym = sym->AsStackSym();
+                    if (stackSym->HasByteCodeRegSlot())
+                    {
+                        Js::RegSlot bytecode = stackSym->GetByteCodeRegSlot();
+                        EnsureBV();
+                        byteCodeUpwardExposedUsed->Set(bytecode);
+                    }
+                }
+            }
+            NEXT_BITSET_IN_SPARSEBV;
+            Assert(block->trackingByteCodeUpwardExposedUsed == nullptr);
+            block->trackingByteCodeUpwardExposedUsed = byteCodeUpwardExposedUsed;
+        }
+        else
+        {
+            // The calculated bytecode upward exposed should be the same between Backward and DeadStore passes
+            Assert(this->tag == Js::DeadStorePhase);
+            if (block->trackingByteCodeUpwardExposedUsed)
+            {
+                // We don't need to track bytecodeUpwardExposeUses if we don't have bailout
+                // We've collected the Backward bytecodeUpwardExposeUses for nothing, oh well.
+                if (this->func->hasBailout)
+                {
+                    Assert(block->byteCodeUpwardExposedUsed);
+                    BVSparse<JitArenaAllocator>* byteCodeUpwardExposedUsed = JitAnew(this->globOpt->tempAlloc, BVSparse<JitArenaAllocator>, this->globOpt->tempAlloc);
+                    FOREACH_BITSET_IN_SPARSEBV(symID, block->byteCodeUpwardExposedUsed)
+                    {
+                        Sym* sym = this->func->m_symTable->Find(symID);
+                        Js::RegSlot bytecode = sym->AsStackSym()->GetByteCodeRegSlot();
+                        byteCodeUpwardExposedUsed->Set(bytecode);
+                    }
+                    NEXT_BITSET_IN_SPARSEBV;
```

Comment:
You could use BVSparse<>::CopyNew to this part.

Reply:
The thing is, it's not really a copy as I have to map SymID to the ByteCode register

Reply:
okay.

---
In reply to: [180938310](https://github.com/Microsoft/ChakraCore/pull/4976#discussion_r180938310) [](ancestors = 180938310)